internal/controller/pipelinerollout_controller.go

```diff
@@ -836,20 +844,17 @@ func updatePipelineSpec(ctx context.Context, restConfig *rest.Config, obj *kuber
 	return kubernetes.UpdateCR(ctx, restConfig, obj, "pipelines")
 }
 
-func pipelineLabels(pipelineRollout *apiv1.PipelineRollout) (map[string]string, error) {
-	var pipelineSpec PipelineSpec
-	labelMapping := map[string]string{
-		common.LabelKeyISBServiceNameForPipeline: "default",
-	}
-	if err := json.Unmarshal(pipelineRollout.Spec.Pipeline.Spec.Raw, &pipelineSpec); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal pipeline spec: %v", err)
-	}
+func pipelineLabels(pipelineSpec *numaflowv1.PipelineSpec) (map[string]string, error) {
+	labelMapping := make(map[string]string)
 	if pipelineSpec.InterStepBufferServiceName != "" {
 		labelMapping[common.LabelKeyISBServiceNameForPipeline] = pipelineSpec.InterStepBufferServiceName
+	} else {
+		labelMapping[common.LabelKeyISBServiceNameForPipeline] = "default"
```

Comment:
maybe doesn't need to be done in this PR, but are we going to add a label for PipelineRollout name? Then we can easily find all Pipelines using that PipelineRollout. I know the Pipeline name will be fixed with the PipelineRollout name, but I think this would enable easy selection both through kubectl and potentially in the code.

Reply:
unless we can do a query on OwnerReference I guess...but just to note that Numaflow includes labels like this

Reply:
FYI, I have just created [this](https://github.com/numaproj/numaplane/issues/245) issue which depends on this

Reply:
yeah, that will be the follow up PR for this.